File: common/src/main/java/com/ultreon/mods/advanceddebug/init/ModOverlays.java
package com.ultreon.mods.advanceddebug.init;

import com.ultreon.mods.advanceddebug.client.menu.DebugGui;
import dev.architectury.injectables.annotations.ExpectPlatform;
import net.minecraft.client.gui.components.Renderable;

@SuppressWarnings("SameParameterValue")
public final class ModOverlays {
    @ExpectPlatform
    public static void registerTop(String name, Renderable overlay) {
        throw new AssertionError();
    }

    public static void registerAll() {
        registerTop("debug_info", DebugGui.get());
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/init/ModOverlays.kt
package com.ultreon.mods.advanceddebug.init

import com.mojang.blaze3d.vertex.PoseStack
import com.ultreon.mods.advanceddebug.client.menu.DebugGui.Companion.get
import dev.architectury.event.events.client.ClientGuiEvent
import net.minecraft.client.gui.components.Renderable

object ModOverlays {
    private val overlays: List<Renderable> = ArrayList()

    @Suppress("UNUSED_PARAMETER")
    private fun registerTop(name: String, overlay: Renderable) {
        ClientGuiEvent.RENDER_HUD.register(ClientGuiEvent.RenderHud { poseStack: PoseStack, tickDelta: Float ->
            overlay.render(
                poseStack,
                Int.MAX_VALUE,
                Int.MAX_VALUE,
                tickDelta
            )
        })
    }

    fun registerAll() {
        registerTop("debug_info", get())
    }
}